src/layers/SelectionLayer.js

```diff
@@ -172,10 +124,8 @@ export default class SelectionLayer extends CompositeLayer {
   }
 
   renderLayers() {
-    const mode = {
-      [SELECTION_TYPE.RECTANGLE]: 'drawRectangle',
-      [SELECTION_TYPE.POLYGON]: 'drawPolygon',
-    }[this.props.selectionType] || 'view';
+    const mode = MODE_MAP[this.props.selectionType] || ViewMode;
+    const modeConfig = MODE_CONFIG_MAP[this.props.selectionType];
```

Comment:
If there were more, we might want just a single MODE_MAP, with destructuring to set multiple variables... but that would probably be gratuitous here?